js-classes/inheritance-es5.js

```diff
@@ -0,0 +1,86 @@
+function Car(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10) {
+  this.name = name;
+  this.model = model;
+  this.year = year;
+  this.color = color;
+  this.maxSpeed = maxSpeed;
+  this.fuelCapacity = fuelCapacity;
+  this.fuelConsumption = fuelConsumption;
+}
+
+Car.prototype.getFullName = function () {
+  return this.name + ' ' + this.model;
+}
+
+Car.prototype.getAge = function () {
+  var currentYear = new Date().getFullYear();
+  var age = currentYear - this.year;
+  return age;
+}
+
+Car.prototype.changeColor = function (color) {
+  if (color == this.color)
+    return 'This car is already ' + color
+  else {
+    this.color = color;
+    return 'New color of this car is ' + color;
+  }
+}
+
+Car.prototype.calculateWay = function (kilometers, fuel) {
+  if (fuel < 10) {
+    return 'Fuel level less than 10L';
+  };
+  var time = (kilometers / this.maxSpeed).toFixed(1);
+  var fuelReq = (kilometers - fuel * 100 / this.fuelConsumption) * this.fuelConsumption / 100;
+  if (kilometers < 100 * (fuel / this.fuelConsumption)) {
+    return 'Trip time is ' + time + '. Refuel is not required';
+  } else {
+    return 'Trip time is ' + time + '. Refuel is required (' + fuelReq + ')';
+  }
+}
+
+function Tesla(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10, panoramicRoof) {
+    Car.call(this, name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10);
```

Comment:
1. не нужно присваивать дефолтные значения здесь, они присваиваются в родительском классе

```suggestion
function Tesla(name, model, year, color, maxSpeed, fuelCapacity, fuelConsumption, panoramicRoof) {
    Car.call(this, name, model, year, color, maxSpeed, fuelCapacity, fuelConsumption);
```

2. можно сделать намного проще, поставив `panoramicRoof` в начало
```suggestion
function Tesla(panoramicRoof, ...baseParams) {
    Car.apply(this, baseParams);
```

Эти пункты касаются всех остальных классов в том числе